### ISSUE TITLE
Multiple Image Upscale From Generated Video

### ISSUE THREAD

**Issue by bar7am** [yuvraj108c/ComfyUI-Upscaler-Tensorrt#43]
I am trying to feed the upscaler node multiple images, but the final output is always the last frame of the batch of images. 

So I'm trying to generate a video. In the screenshot below, I have the VAE decode after the sampler. On top, I'm previewing the frames from the sampler directly, and below I'm sending the images from VAE decode into the ComfyUI-Upscaler-Tensorrt node, then previewing the output from the upscaler. 

As you can see, the number of frames is correct, but it's taken the last frame from the generated output only, and upscaled that over and over. 

![Image](https://github.com/user-attachments/assets/be559b67-6f3f-4359-8bfa-528c102b9f22)

I'm not seeing any errors in the console. 

![Image](https://github.com/user-attachments/assets/8c68b1fd-a77d-4152-b791-b32b275da246)

Am I doing this wrong? Can this upscaler accept a bunch of images and upscale each one?

**Comment by bar7am** [comment]
You were right about the vram issues, unless I scaled the input frames down, the spike after upscale caused OOMs

![Image](https://github.com/user-attachments/assets/d52ea308-521b-4af0-bad7-b29127ca6ea4)

I modified `upscaler_tensorrt` to try and fix the tensor accumulation in vram. 

Removed gpu based accumulation list / replaced with pre-allocated cpu tensor
`final_output = torch.empty((B, C, final_height, final_width), 
                           dtype=torch.float32, 
                           device="cpu")

for i, img in enumerate(images_list):
    result = upscaler_trt_model.infer({"input": img}, cudaStream)`

Changed from batch to per frame resizing
`if H*4 != final_height or W*4 != final_width:
    current_output = torch.nn.functional.interpolate(
        current_output, 
        size=(final_height, final_width),
        mode='bicubic',
        antialias=True
    )`

Immediate CPU offloading of processed frames
`final_output[i] = current_output.detach().cpu()`

Memory management
`del result, current_output
torch.cuda.synchronize()  # Ensure GPU operations complete

if i % 10 == 0:
    torch.cuda.empty_cache()`

Changed return
`upscaler_trt_model.reset()

result = final_output.to(mm.intermediate_device()).permute(0, 2, 3, 1)

del final_output
mm.soft_empty_cache()

return (result,)`

Now the vram is stable on my end without too much of a performance hit. 

![Image](https://github.com/user-attachments/assets/ddf7bf39-ce4c-40e0-ba3e-ac677fa55891)

After a bit of testing, this implementation seems to work fine.

**Comment by yuvraj108c** [comment]
Done!
Thanks for the help.

**Comment by bar7am** [comment]
> Done! Thanks for the help.

👍
